File: DynamicForm/DynamicForm/Models/Question.swift
//
//  Question.swift
//  DynamicForm
//
//  Created by Jess Melanson on 5/19/24.
//

import Foundation

struct Question: Codable {
  let uuid: String
  let prompt: String
  let answers: [String]
  let nextItemUuids: [String]
  let nextItemTypes: [String]
}

File: models/question.js
import { Schema, model } from 'mongoose';

const questionSchema = new Schema({
  uuid: {
    type: String,
    required: true
  },
  prompt: {
    type: String,
    required: true
  },
  answers: {
    type: [String],
    required: true
  },
  nextItemUuids: {
    type: [String],
    required: true
  },
  nextItemTypes: {
    type: [String],
    required: true
  }
});

const Question = model('Question', questionSchema)

export default Question;